internal/catalog/internal/controllers/endpoints/bound.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package endpoints
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+// GetBoundIdentities returns the unique list of workload identity references
+// encoded into a data-bearing status condition on a Service resource by the
+// endpoints controller.
+//
+// This allows a controller to skip watching ServiceEndpoints (which is
+// expensive) to discover this data.
+func GetBoundIdentities(typ *pbresource.Type, res *pbresource.Resource) []*pbresource.Reference {
```

Comment:
Don't we always want `typ`to be workload identity?

Reply:
Yes, but then we need to have a `catalog->pbauth` dependency.

Reply:
It might be worth adding a comment on this. If not, I'm sure someone will remove this argument in the future.